source/common/request_source_plugin_impl.cc

```diff
@@ -0,0 +1,106 @@
+
+#include "common/request_impl.h"
+#include "common/request_source_impl.h"
+
+namespace Nighthawk {
+
+std::string DummyRequestSourceConfigFactory::name() const {
+  return "nighthawk.dummy-request-source-plugin";
+}
+
+Envoy::ProtobufTypes::MessagePtr DummyRequestSourceConfigFactory::createEmptyConfigProto() {
+  return std::make_unique<nighthawk::request_source::DummyPluginRequestSourceConfig>();
+}
+
+RequestSourcePluginPtr
+DummyRequestSourceConfigFactory::createRequestSourcePlugin(const Envoy::Protobuf::Message& message,
+                                                           Envoy::Api::Api& api) {
+  const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+  nighthawk::request_source::DummyPluginRequestSourceConfig config;
+  Envoy::MessageUtil::unpackTo(any, config);
+  return std::make_unique<DummyRequestSourcePlugin>(config, api);
+}
+
+REGISTER_FACTORY(DummyRequestSourceConfigFactory, RequestSourcePluginConfigFactory);
+
+DummyRequestSourcePlugin::DummyRequestSourcePlugin(
+    const nighthawk::request_source::DummyPluginRequestSourceConfig& config, Envoy::Api::Api& api)
+    : RequestSourcePlugin{api}, dummy_value_{config.has_dummy_value()
+                                                 ? config.dummy_value().value()
+                                                 : std::numeric_limits<double>::infinity()} {}
+RequestGenerator DummyRequestSourcePlugin::get() {
+  RequestGenerator request_generator = []() {
+    Envoy::Http::RequestHeaderMapPtr header = Envoy::Http::RequestHeaderMapImpl::create();
+    auto returned_request_impl = std::make_unique<RequestImpl>(std::move(header));
+    return returned_request_impl;
+  };
+  return request_generator;
+}
+void DummyRequestSourcePlugin::initOnThread() {}
+
+std::string FileBasedRequestSourceConfigFactory::name() const {
+  return "nighthawk.file-based-request-source-plugin";
+}
+
+Envoy::ProtobufTypes::MessagePtr FileBasedRequestSourceConfigFactory::createEmptyConfigProto() {
+  return std::make_unique<nighthawk::request_source::FileBasedPluginRequestSourceConfig>();
+}
+
+RequestSourcePluginPtr FileBasedRequestSourceConfigFactory::createRequestSourcePlugin(
+    const Envoy::Protobuf::Message& message, Envoy::Api::Api& api) {
+  const auto& any = dynamic_cast<const Envoy::ProtobufWkt::Any&>(message);
+  nighthawk::request_source::FileBasedPluginRequestSourceConfig config;
+  Envoy::MessageUtil::unpackTo(any, config);
+  return std::make_unique<FileBasedRequestSourcePlugin>(config, api);
+}
+
+REGISTER_FACTORY(FileBasedRequestSourceConfigFactory, RequestSourcePluginConfigFactory);
+
+FileBasedRequestSourcePlugin::FileBasedRequestSourcePlugin(
+    const nighthawk::request_source::FileBasedPluginRequestSourceConfig& config,
+    Envoy::Api::Api& api)
+    : RequestSourcePlugin{api}, uri_(config.uri()), file_path_(config.file_path()) {
+  Envoy::MessageUtil util;
+  util.loadFromFile(file_path_, options_list_, Envoy::ProtobufMessage::getStrictValidationVisitor(),
```

Comment:
Looking at this, I wonder what would happen here when reading super large files.  If the use case is reading files that fit into random access memory (also when considering running with lots of workers), then maybe renaming this class to something to reflect that would be good?

Reply:
Also -- I don't remember, but this doesn't happen concurrently right? If it does, should we be concerned about loadFromFile using exclusive file-modes? 

Also, independently -- I am curious, when multiple workers are involved, should they all start at 0 in the file / have a full copy? Or can they share that memory? 
Or could/should they offsets, each taking an equal chunk of the file for reply purposes? In that case we might save some memory by storing only what a worker needs.

Reply:
The load from file concern is something I am addressing, but the use case for this is indeed for smaller files, since this is mostly a small plugin to act as a test case for the change.